packages/files-ui/cypress/tests/file-sharing-spec.ts

```diff
@@ -166,5 +170,152 @@ describe("File Sharing", () => {
+      homePage.uploadFile("../fixtures/uploadedFiles/logo.png")
+      homePage.fileItemName().invoke("text").as("fileName")
+
+      // share file to newly created share folder
+      homePage.fileItemKebabButton().click()
+      homePage.shareMenuOption().click()
+      shareFileModal.body().should("be.visible")
+      shareFileModal.shareNameInput().type(sharedFolderName)
+      // click to unmark checkbox, selected by default
+      shareFileModal.keepOriginalFilesCheckbox().click()
+      shareFileModal.moveOverButton().click()
+      sharingProgressToast.body().should("be.visible")
+      shareSuccessToast.body().should("be.visible")
+      shareSuccessToast.closeButton().click()
+      editSharedFolderModal.closeButton().click()
+
+      // ensure the file did not remain in the drive after moving
+      homePage.fileItemRow().should("have.length", 0)
+
+      // go to the new share and ensure the file was moved there
+      navigationMenu.sharedNavButton().click()
+      sharedPage.sharedFolderItemName().contains(sharedFolderName)
+        .should("be.visible")
+        .dblclick()
+      sharedPage.fileItemRow().should("have.length", 1)
+
+      // ensure file name of moved file is correct
+      cy.get("@fileName").then(($fileName) => {
+        sharedPage.fileItemName().contains(`${$fileName}`).should("be.visible")
+      })
+    })
+
+    it("can copy a file to a pre-existing shared folder and preserve original", () => {
+      cy.web3Login({ deleteShareBucket: true, clearCSFBucket: true })
+
+      navigationMenu.sharedNavButton().click()
+      sharedPage.createSharedFolder()
+
+      // upload a file and store it's name as cypress alias
+      navigationMenu.homeNavButton().click()
+      homePage.uploadFile("../fixtures/uploadedFiles/logo.png")
+      homePage.fileItemName().invoke("text").as("fileName")
+
+      // share file to newly created share folder
+      homePage.fileItemKebabButton().click()
+      homePage.shareMenuOption().click()
+      shareFileModal.body().should("be.visible")
+      shareFileModal.selectFolderInput().click()
+      shareFileModal.selectFolderInput().type(`${sharedFolderName}{enter}`)
+      shareFileModal.copyOverButton().click()
+      sharingProgressToast.body().should("be.visible")
+      shareSuccessToast.body().should("be.visible")
+      shareSuccessToast.closeButton().click()
+
+      // ensure file remained in drive after copying
+      homePage.fileItemRow().should("have.length", 1)
+
+      // go to the new share and ensure file was copied there
+      navigationMenu.sharedNavButton().click()
+      sharedPage.sharedFolderItemName().contains(sharedFolderName)
+        .should("be.visible")
+        .dblclick()
+      sharedPage.fileItemRow().should("have.length", 1)
+
+      // ensure file name of copied file is correct
+      cy.get("@fileName").then(($fileName) => {
+        sharedPage.fileItemName().contains(`${$fileName}`).should("be.visible")
```

Comment:
A little more elegant IMHO. Feel free to ignore if you feel it's too complex
```suggestion
      cy.get<string>("@fileName").then((fileName) => {
        sharedPage.fileItemName().contains(fileName).should("be.visible")
```